### PR TITLE
Order framework

### PR DESCRIPTION
**Pull request by santhosh-mankala-volusion**
User should be able to add a product and make all changes like add products, update shipping,update coupons and also edit any exisitng orders. 

User will not be able to submit a new order, but can make changes in pending state.

### CODE REVIEW

File: MozuAndroidInStoreAssistant/app/src/main/java/com/mozu/mozuandroidinstoreassistant/app/layout/order/NewOrderCouponLayout.java [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+                @Override
+                public void run() {
+                    mSpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+                        @Override
+                        public void onItemSelected(AdapterView<?> adapterView, View view, int position, long l) {
+                            if (position == 0)
+                                return;
+                            final String couponSelected = (String) adapterView.getItemAtPosition(position);
+                            progressBar.setVisibility(View.VISIBLE);
+                            NewOrderManager.getInstance().getApplyCouponObservable(mTenantId, mSiteId, order.getId(), couponSelected)
```
Comment (chris-pound-volusion) [inline MozuAndroidInStoreAssistant/app/src/main/java/com/mozu/mozuandroidinstoreassistant/app/layout/order/NewOrderCouponLayout.java]: does this need to be unsubscribed manually

File: MozuAndroidInStoreAssistant/app/src/main/java/com/mozu/mozuandroidinstoreassistant/app/order/NewOrderCreateFragment.java [hunk 2]
```diff
@@ -129,41 +201,204 @@ public void afterTextChanged(Editable editable) {
...
+            outProduct.setProperties(properties);
+        }
+
+        outProduct.setUpc(inProduct.getUpc());
+        outProduct.setVariationProductCode(inProduct.getVariationProductCode());
+
+        return outProduct;
     }
 
+    public static com.mozu.api.contracts.commerceruntime.products.ProductProperty convertProperties(
```
Comment (chris-pound-volusion) [inline MozuAndroidInStoreAssistant/app/src/main/java/com/mozu/mozuandroidinstoreassistant/app/order/NewOrderCreateFragment.java]: why the full package name?

File: MozuAndroidInStoreAssistant/app/src/main/java/com/mozu/mozuandroidinstoreassistant/app/order/NewOrderItemEditFragment.java [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+                .subscribe(new Subscriber<List<FulfillmentInfo>>() {
+                    @Override
+                    public void onCompleted() {
+                        fulfillmentSpinnerProgress.setVisibility(View.GONE);
+                        productSave.setEnabled(true);
+                    }
+
+                    @Override
+                    public void onError(Throwable e) {
+                        Toast.makeText(getActivity(), e.getMessage(), Toast.LENGTH_SHORT).show();
```
Comment (chris-pound-volusion) [inline MozuAndroidInStoreAssistant/app/src/main/java/com/mozu/mozuandroidinstoreassistant/app/order/NewOrderItemEditFragment.java]: there's a standard error dialog I made.ErrorMessageAlertDialog. I think we should use that for onError fails. Thoughts?

File: MozuAndroidInStoreAssistant/app/src/main/java/com/mozu/mozuandroidinstoreassistant/app/order/NewOrderItemEditFragment.java [hunk 4]
```diff
@@ -0,0 +1,473 @@
...
+                        updatedProductQuantityVal = null;
+
+                    }
+                    if (updatedProductQuantityVal != null || updatedFulFillmentType != null) {
+                        AndroidObservable.bindFragment(NewOrderItemEditFragment.this, NewOrderManager
+                                .getInstance()
+                                .getOrderItemUpdateQuantityObservable(mTenantId, mSiteId, mOrderItem, mOrderId, updatedProductQuantityVal == null ? null : Integer.valueOf(updatedProductQuantityVal), updatedFulFillmentType))
+                                .subscribeOn(Schedulers.io())
+                                .observeOn(AndroidSchedulers.mainThread())
+                                .subscribe(new Subscriber<Order>() {
```
Comment (chris-pound-volusion) [inline MozuAndroidInStoreAssistant/app/src/main/java/com/mozu/mozuandroidinstoreassistant/app/order/NewOrderItemEditFragment.java]: these subscribers are duplicates anyway to put them in a variable or method?

File: MozuAndroidInStoreAssistant/app/src/main/java/com/mozu/mozuandroidinstoreassistant/app/order/NewOrderCreateFragment.java (hunk 2 above)
Comment (santhosh-mankala-volusion) [inline MozuAndroidInStoreAssistant/app/src/main/java/com/mozu/mozuandroidinstoreassistant/app/order/NewOrderCreateFragment.java]: Here we are trying to convert from runtime property to commerce product property which have same class names. hence the full package name to differentiate both

File: MozuAndroidInStoreAssistant/app/src/main/java/com/mozu/mozuandroidinstoreassistant/app/order/NewOrderItemEditFragment.java (hunk 3 above)
Comment (santhosh-mankala-volusion) [inline MozuAndroidInStoreAssistant/app/src/main/java/com/mozu/mozuandroidinstoreassistant/app/order/NewOrderItemEditFragment.java]: Yes. That makes sense

**Pull request merged**